tests/bsim/bluetooth/mesh/src/test_dfu.c

```diff
@@ -1034,6 +1035,186 @@ static void test_cli_fail_on_persistency(void)
+				   &dfu_cli_xfer.xfer);
+	if (err) {
+		FAIL("DFU Client send failed (err: %d)", err);
+	}
+
+	if (k_sem_take(&dfu_ended, K_SECONDS(200))) {
+		FAIL("Firmware transfer failed");
+	}
+
+	for (i = 0; i < dfu_targets_cnt; i++) {
+		ASSERT_EQUAL(BT_MESH_DFU_ERR_METADATA, dfu_cli_xfer.targets[i].status);
+		ASSERT_EQUAL(BT_MESH_DFU_PHASE_IDLE, dfu_cli_xfer.targets[i].phase);
+	}
+
+	/* `lost_target` cb must be called on all targets */
+	ASSERT_EQUAL(0, lost_targets_rem());
+
+	PASS();
+}
+
+static void test_cli_all_targets_lost_on_caps_get(void)
+{
+	int err, i;
+
+	expect_fail = true;
+
+	for (i = 1; i <= dfu_targets_cnt; i++) {
+		(void)target_srv_add(TARGET_ADDR + i, true);
+	}
+
+	cli_common_fail_on_init();
+
+	err = bt_mesh_dfu_cli_send(&dfu_cli, &dfu_cli_xfer.inputs, &dummy_blob_io,
+				   &dfu_cli_xfer.xfer);
+	if (err) {
+		FAIL("DFU Client send failed (err: %d)", err);
+	}
+
+	if (k_sem_take(&dfu_ended, K_SECONDS(200))) {
+		FAIL("Firmware transfer failed");
+	}
+
+	for (i = 0; i < dfu_targets_cnt; i++) {
+		ASSERT_EQUAL(BT_MESH_DFU_ERR_INTERNAL, dfu_cli_xfer.targets[i].status);
+		ASSERT_EQUAL(BT_MESH_DFU_PHASE_TRANSFER_ACTIVE,
+			     dfu_cli_xfer.targets[i].phase);
+	}
+
+	/* `lost_target` cb must be called on all targets */
+	ASSERT_EQUAL(0, lost_targets_rem());
+
+	PASS();
+}
+
+static void test_cli_all_targets_lost_on_update_get(void)
+{
+	int err, i;
+
+	expect_fail = true;
+
+	for (i = 1; i <= dfu_targets_cnt; i++) {
+		(void)target_srv_add(TARGET_ADDR + i, true);
+	}
+
+	cli_common_fail_on_init();
+
+	err = bt_mesh_dfu_cli_send(&dfu_cli, &dfu_cli_xfer.inputs, &dummy_blob_io,
+				   &dfu_cli_xfer.xfer);
+	if (err) {
+		FAIL("DFU Client send failed (err: %d)", err);
+	}
+
+	if (k_sem_take(&dfu_ended, K_SECONDS(200))) {
+		FAIL("Firmware transfer failed");
+	}
+
+	for (i = 0; i < dfu_targets_cnt; i++) {
+		ASSERT_EQUAL(BT_MESH_DFU_ERR_INTERNAL, dfu_cli_xfer.targets[i].status);
+		ASSERT_EQUAL(BT_MESH_DFU_PHASE_TRANSFER_ACTIVE,
+			     dfu_cli_xfer.targets[i].phase);
+	}
+
+	/* `lost_target` cb must be called on all targets */
+	ASSERT_EQUAL(0, lost_targets_rem());
+
+	PASS();
+}
+
+static void test_cli_all_targets_lost_on_verify(void)
+{
+	int err, i;
+
+	expect_fail = true;
+
+	for (i = 1; i <= dfu_targets_cnt; i++) {
+		(void)target_srv_add(TARGET_ADDR + i, true);
+	}
+
+	cli_common_fail_on_init();
+
+	err = bt_mesh_dfu_cli_send(&dfu_cli, &dfu_cli_xfer.inputs, &dummy_blob_io,
+				   &dfu_cli_xfer.xfer);
+	if (err) {
+		FAIL("DFU Client send failed (err: %d)", err);
+	}
+
+	if (k_sem_take(&dfu_ended, K_SECONDS(200))) {
+		FAIL("Firmware transfer failed");
+	}
+
+	for (i = 0; i < dfu_targets_cnt; i++) {
+		ASSERT_EQUAL(BT_MESH_DFU_ERR_WRONG_PHASE, dfu_cli_xfer.targets[i].status);
+		ASSERT_EQUAL(BT_MESH_DFU_PHASE_VERIFY_FAIL, dfu_cli_xfer.targets[i].phase);
+	}
+
+	/* `lost_target` cb must be called on all targets */
+	ASSERT_EQUAL(0, lost_targets_rem());
+
+	PASS();
+}
+
+static void test_cli_all_targets_lost_on_apply(void)
```

Comment:
I see the body of this function and the 4 functions above are almost the same. Could you make a parametrized function that gets the difference as an argument instead of copy-paste of the body?